lib/index.js

```diff
@@ -432,125 +432,137 @@ Utils.parseDateTimeField = function(field, options) {
     return formatParsedObject('date', field, isValid, parsed);
   };
 
+  Utils.Logger.debug('Utils.parseDateTimeField: params:', field, options);
+
   // Copy options
   options = _.assign({}, options);
 
   // Default timezone
-  var timezone = options.timezone || 'UTC';
-  var isUTC = timezone.toLowerCase() === 'utc';
+  var timezone = options.timezone || moment.tz.guess();
+  delete options.timezone;
 
   // Lookup Locale. Check options, then look up based on timezone. Default to 'en'
   options.locale = options.locale || TIMEZONE_TO_LOCALE[timezone] || 'en';
 
   // If invalid
   if (moment.tz.zone(timezone) === null) {
-    Utils.Logger.warn('Utils.parseDateTimeField - Timezone "' + timezone + '" is invalid. Assuming UTC');
-    timezone = 'UTC';
-  }
-
-  // This special case is here to get around a bug in SugarDate. Issue: #582
-  options.fromUTC = field === 'now' ? false : isUTC;
-
-  // Help SugarDate be aware of timezones
-  var previousNewDateInternal;
-  if (timezone) {
-    previousNewDateInternal = SugarDate.getOption('newDateInternal');
-    SugarDate.setOption('newDateInternal', function() {
-      return moment().tz(timezone).toDate();
-    });
+    var tz = moment.tz.guess() || 'UTC';
+    Utils.Logger.warn('Utils.parseDateTimeField - Timezone "' + timezone + '" is invalid. Assuming ' + tz);
+    timezone = tz;
```

Comment:
We're changing `timezone` here but the `options.locale` was already decided above, with the old timezone. Does line 445 need to be moved to below this `if` block to take the new timezone into account?